### PR TITLE
add tests for the node resize feature and properties panel

### PR DESCRIPTION
**Pull request by jomarko**
As part of this PR we add:
- Tests of the resize feature. We select node, using its `resize-handle` element we increase its size and then we check the size property in the properties panel
- We change node properties and check the properties panel effect
- We change the diagram properties and check the effect in the properties panel
- We change font properties when multiple nodes are selected

### CODE REVIEW

File: packages/dmn-editor/src/propertiesPanel/FontOptions.tsx [hunk 1]
```diff
@@ -374,22 +373,24 @@ export function FontOptions({ startExpanded, nodeIds }: { startExpanded: boolean
...
+                selections={fontFamily ?? ""}
+                isDisabled={false}
+                maxHeight={inViewTimezoneSelect.maxHeight}
+                direction={inViewTimezoneSelect.direction}
+              >
+                {WEBSAFE_FONTS_LIST.map((fontName, index) => (
+                  <SelectOption key={index} value={fontName} style={{ fontFamily: fontName }} />
+                ))}
+              </Select>
+            </div>
```
Comment (jomarko) [inline packages/dmn-editor/src/propertiesPanel/FontOptions.tsx:393]: We can not simply add `data-testid` int `Select`, patternfly for some reason ignores that attribute.

**Review by ljmotta — CHANGES_REQUESTED** [review]
Alright, the tests are looking good, but it's still missing a few ones. Some of them I've added inline on the test files. We also need to test the documentation links with more cases. Add one more and change orders, for example. We also need to test the reset style button, for now, we can only test with the node font. The documentation links can be added to a different file, as it's not node specific.

Great job on the resize tests. But they are missing a few cases too:
 - Resize with the snapping on?
 - Change the snapping values?
 - Resize on top of another node?

Now, regarding the fixtures, I don't think they're looking good. The class inheritance doesn't fit in some cases, making it necessary to add `throw new Error` methods. The way to solve this is with class composition. Instead of inheritance, we pass the instance or create the instance on the constructor and attribute it to one property. This way the class will have only the necessary methods. Also, I don't think this would be good either. We can't be sure the nodes will always share those properties, and maybe the best way is to replicate the logic on each class, removing the coupling altogether. Duplicating the code isn't a bad thing in this case, as it'll remove the complexity of creating the abstraction.

File: packages/dmn-editor/src/propertiesPanel/FontOptions.tsx (hunk 1 above)
Comment (jomarko) [reply packages/dmn-editor/src/propertiesPanel/FontOptions.tsx:393]: Probably we can not, `<React.Fragment data-testid={"xyz"}>` has no effect for the produced html.  The `data-testid` is not in the generated html at all.

File: packages/dmn-editor/tests/e2e/drgElements/changeMultipleNodesProperties.spec.ts
[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeMultipleNodesProperties.spec.ts:1]: moved one level up

[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeMultipleNodesProperties.spec.ts:1]: Sorry, I do not get it, as I tried to explain, I am not able to write a test for a color change using playwright. Could you please help me and advise how to write such a test? I am not able to handle the color picker we use in KIE tooling.

File: packages/dmn-editor/tests/e2e/drgElements/changeDiagramProperties.spec.ts
[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeDiagramProperties.spec.ts:1]: tests added

File: packages/dmn-editor/tests/e2e/drgElements/changeDecisionServiceProperties.spec.ts
[the remark this answers is not in the document]
Comment (jomarko) [reply packages/dmn-editor/tests/e2e/drgElements/changeDecisionServiceProperties.spec.ts:1]: I agree we need such tests, will add them as part of https://github.com/ljmotta/kie-tools/pull/29

**Comment by jomarko** [comment]
replaced by https://github.com/apache/incubator-kie-tools/pull/2239

**Pull request closed without merging**